src/agent/src/agent.cpp

```diff
@@ -43,6 +59,52 @@ void Agent::Run()
         [this]([[maybe_unused]] const std::string& response)
         { PopMessagesFromQueue(m_messageQueue, MessageType::STATELESS); }));
 
+    m_taskManager.EnqueueTask(m_commandHandler.ProcessCommandsFromQueue<command_store::Command>(
+        [this]() -> std::optional<command_store::Command>
+        {
+            if (m_agentQueue.isEmpty(MessageType::COMMAND))
+            {
+                return std::nullopt;
+            }
+            Message m = m_agentQueue.getNext(MessageType::COMMAND);
+
+            std::cout << "COMMAND retrieved from Queue:" << "\n";
+
+            int id = std::stoi(GetJsonValue(m.data, "id"));
+            std::string module = m.moduleName;
+            std::string command = GetJsonValue(m.data, "command");
+            std::string parameters = GetJsonValue(m.data, "parameters");
+            std::string result = GetJsonValue(m.data, "error");
+            command_store::Status status = command_store::Status::IN_PROGRESS;
+            command_store::Command cmd(id, module, command, parameters, result, status);
+
+            // nlohmann::json jdata = m.data.at(0);
+            // for (auto& [key, value] : jdata.items())
+            // {
+            //     std::cout << key << ": " << value << "\n";
+            // }
+
+            // std::cout << "Data inside data:" << "\n";
+
+            // nlohmann::json jdataData = m.data.at(0).at("data");
+            // for (auto& [key, value] : jdataData.items())
+            // {
+            //     std::cout << key << ": " << value << std::endl;
+            // }
+
+            return cmd;
+        },
+        [this]() -> bool
```

Comment:
Same, here we can create and use a similar function than `PopMessagesFromQueue`. Maybe `PopCommandFromQueue`?

Reply:
Done